### ISSUE TITLE
GATConv but for Dense Networks

### ISSUE THREAD

**Issue by tarungog** [pyg-team/pytorch_geometric#1141]
## 🚀 Feature

Hi, I have a very simple request. I've seen that you've made dense formulations of a couple of the graph convolution types and graph pooling operators. Could you make the same thing for GATConv (and NNConv for that matter). If you don't have the bandwidth, just give me some pointers as to how to make this thing myself.
## Motivation

I'm basically using GATConv as a full self-attention network (all nodes connected).

**Comment by rusty1s** [comment]
That is a reasonable request, and I would be happy to include it into PyG. It would be great if you could work on this. For implementing, I would mostly follow the [`DenseGraphConv`](https://github.com/rusty1s/pytorch_geometric/blob/master/torch_geometric/nn/dense/dense_graph_conv.py) implementation. For GAT, you need to calculate attention weights first (with a masked softmax) before applying the matrix multiplication.

**Comment by zhxchd** [comment]
Hi there, my project also involves GAT operators on dense adjacency matrices. If this wasn't followed by anyone for now, I'll try to work on this for a `DenseGATConv` operator.

**Comment by rusty1s** [comment]
Super :)
